src/main/scala/io/keen/client/scala/Client.scala

```diff
@@ -381,6 +408,170 @@ trait Writer extends AccessLevel {
+                error(s"""$ex""")
+              }
+            }
+          }
+        }, 1, sendIntervalSeconds.toLong, TimeUnit.SECONDS)
+
+        Some(tp)
+    }
+
+  }
+
+  /**
+   * Sends all queued events, removing events from the queue as events are successfully sent.
+   */
+  def sendQueuedEvents(): Unit = {
+
+    val handleMap: TrieMap[String, ListBuffer[Long]] = eventStore.getHandles(projectId)
+    val handles: ListBuffer[Long] = ListBuffer.empty[Long]
+    val events: ListBuffer[String] = ListBuffer.empty[String]
+
+    // iterate over all of the event handles in the queue, by collection
+    for((collection, eventHandles) <- handleMap) {
+      // get each event, and its handle, then add it to a buffer so we can group the events 
+      // into smaller batches
+      for(handle <- eventHandles) {
+        handles += handle
+        events += eventStore.get(handle)
+      }
+
+      // group handles separately so we can use them to remove events from the queue once they've
+      // been succesfully added
+      val handleGroup: List[ListBuffer[Long]] = handles.grouped(batchSize).toList
+
+      // group the events by batch size, then publish them
+      for((batch, index) <- events.grouped(batchSize).zipWithIndex) {
+        // publish this batch
+        var response = Await.result(
+          addEvents(s"""{"$collection": [${batch.mkString(",")}]}"""),
+          DurationInt(batchTimeout).seconds
+        )
+
+        // handle addEvents responses properly
+        response.statusCode match {
+          case 200 | 201 => {
+            // log success
+            info(s"""${response.statusCode} ${response.body} | Sent ${batch.size} queued events""")
+
+            // remove all of the handles for this batch
+            for(handle <- handleGroup(index)) {
+              eventStore.remove(handle)
+            }
+
+            // log removal
+            info(s"""Removed ${handleGroup(index).size} events from the queue""")
+          }
+          case _ => {
+            // log but DO NOT remove events from queue
+            error(s"""${response.statusCode} ${response.body} | Failed to send ${batch.size} queued events""") 
+          }
+        }
+      }
+    }
+
+  }
+
+  /**
+   * Asynchronously sends all queued events.
+   */
+  def sendQueuedEventsAsync(): Unit = {
+
+    // use a thread pool for our async thread so we can use daemon threads
+    val tp = Executors.newSingleThreadExecutor(new ClientThreadFactory)
+
+    // send our queued events in a separate thread
+    tp.execute(new Runnable {
+      def run: Unit = {
+        sendQueuedEvents()
+      }
+    })
+
+  }
+
+  /**
+   * Safely shuts down `scheduledThreadPool` before sending all events remaining in `eventStore`.
+   */
+  override def shutdown() = {
+
+    // safely terminate the scheduled thread pool
+    scheduledThreadPool match {
+      case Some(stp) =>
+        stp.shutdown
+        stp.awaitTermination(shutdownDelay.toLong, TimeUnit.SECONDS) match {
```

Comment:
Pretty sure this is a timeout, not a delay.  Is there any reason it is not in the reference.conf?